Gems/Prefab/PrefabBuilder/PrefabGroup/PrefabGroupBehavior.cpp

```diff
@@ -51,19 +81,290 @@ namespace AZ::SceneAPI::Behaviors
         {
             BindToCall(&ExportEventHandler::PrepareForExport);
             AZ::SceneAPI::SceneCore::ExportingComponent::Activate();
+            Events::AssetImportRequestBus::Handler::BusConnect();
         }
 
         ~ExportEventHandler()
         {
+            Events::AssetImportRequestBus::Handler::BusDisconnect();
             AZ::SceneAPI::SceneCore::ExportingComponent::Deactivate();
         }
 
         Events::ProcessingResult PrepareForExport(Events::PreExportEventContext& context)
         {
             return m_preExportEventContextFunction(context);
         }
+
+        // AssetImportRequest
+        Events::ProcessingResult UpdateManifest(Containers::Scene& scene, ManifestAction action, RequestingApplication requester) override;
     };
 
+    Events::ProcessingResult PrefabGroupBehavior::ExportEventHandler::UpdateManifest(
+        Containers::Scene& scene,
+        ManifestAction action,
+        [[maybe_unused]] RequestingApplication requester)
+    {
+        if (action != Events::AssetImportRequest::ConstructDefault)
+        {
+            return Events::ProcessingResult::Ignored;
+        }
+
+        // this toggle makes constructing default mesh groups and a prefab optional
+        if (AZ::SettingsRegistryInterface* settingsRegistry = AZ::SettingsRegistry::Get(); settingsRegistry)
+        {
+            bool createDefaultPrefab = true;
+            settingsRegistry->Get(createDefaultPrefab, s_PrefabGroupBehaviorCreateDefaultKey);
+            if (createDefaultPrefab == false)
+            {
+                return Events::ProcessingResult::Ignored;
+            }
+        }
+
+        using MeshTransformPair = AZStd::pair<Containers::SceneGraph::NodeIndex, Containers::SceneGraph::NodeIndex>;
+        using MeshTransformEntry = AZStd::pair<Containers::SceneGraph::NodeIndex, MeshTransformPair>;
+        using MeshTransformMap = AZStd::unordered_map<Containers::SceneGraph::NodeIndex, MeshTransformPair>;
+        MeshTransformMap meshTransformMap;
+
+        auto graph = scene.GetGraph();
+        const auto view = Containers::Views::MakeSceneGraphDownwardsView<Containers::Views::BreadthFirst>(
+            graph,
+            graph.GetRoot(),
+            graph.GetContentStorage().cbegin(),
+            true);
+
+        if (view.begin() == view.end())
+        {
+            return Events::ProcessingResult::Ignored;
```

Comment:
I think the customer might be making non-mesh data scenes like animation scenes which should not show warnings or errors, right?

Reply:
Yeah, that doesn't sound like a warning or error. A log message would be helpful "Scene contains no meshes, skipping prefab generation" in case it's not intentional, but isn't necessary. I keep going back and forth on how much information we should have in our logs. If you don't think it's necessary to have a log then I think we can hold off for now, and wait for feedback.